spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -187,33 +190,40 @@ protected void doStart() {
 	@Override
 	protected void doStop() {
 		this.topicLock.lock();
-		String[] topics = getTopic();
+		initializeMqttAsyncClientIfRequired();
+		String[] topics = getTopicNames();
 		try {
 			this.mqttClient.unsubscribe(topics).waitForCompletion(getCompletionTimeout());
 			this.mqttClient.disconnect().waitForCompletion(getCompletionTimeout());
-		}
-		catch (MqttException ex) {
+		} catch (MqttException ex) {
```

Comment:
That's not correct code style.

Please, run `./gradlew clean :spring-integration-mqtt:check` to see all the Checkstyle violations.